recipes/system-probe.rb

```diff
@@ -0,0 +1,49 @@
+include_recipe 'datadog::dd-agent'
+
+# Set the system-probe agent service enable or disable
+agent_enable = node['datadog']['system_probe']['enabled'] ? :enable : :disable
+# Set the correct Agent startup action
+agent_start = node['datadog']['system_probe']['enabled'] ? :start : :stop
+
+#
+# Configures system-probe agent
+#
+if node['datadog']['agent6']
+  include_recipe 'datadog::_agent6_config'
+
+  system_probe_config_file = ::File.join(node['datadog']['agent6_config_dir'], 'system-probe.yaml')
+  template system_probe_config_file do
+    source 'system_probe.yaml.erb'
+    variables(
+      enabled: node['datadog']['system_probe']['enabled'],
+      sysprobe_socket: node['datadog']['system_probe']['sysprobe_socket'],
+      debug_port: node['datadog']['system_probe']['debug_port'],
+      bpf_debug: node['datadog']['system_probe']['bpf_debug']
+    )
+    owner 'dd-agent'
+    group 'root'
+    mode '640'
+    notifies :restart, 'service[datadog-agent-sysprobe]', :delayed unless agent_start == false
+    # since process-agent collects network info through system-probe, enabling system-probe should also restart process-agent
+    notifies :restart, 'service[datadog-agent]', :delayed unless agent_start == false
+  end
+else
+  raise "system-probe agent does not run in agent5 environment"
+end
+
+# Common configuration
+service_provider = nil
+if node['datadog']['agent6'] &&
+   (((node['platform'] == 'amazon' || node['platform_family'] == 'amazon') && node['platform_version'].to_i != 2) ||
```

Comment:
Is there a better way to detect upstart?  This seems  like it could be brittle

Reply:
I copied this from main datadog-agent for detection, I'm not sure what's a better way.

Reply:
Can we turn this into a helper function that both recipes can call